vertical-pod-autoscaler/initializer/initializer.go

```diff
@@ -0,0 +1,347 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"log"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	watch "k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/apis/admissionregistration/v1alpha1"
+
+	"time"
+
+	kubeclient "k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	admissionregistrationv1alpha1 "k8s.io/kubernetes/pkg/client/clientset_generated/clientset/typed/admissionregistration/v1alpha1"
+
+	"github.com/golang/glog"
+	"k8s.io/autoscaler/vertical-pod-autoscaler/apimock"
+	recommender "k8s.io/autoscaler/vertical-pod-autoscaler/recommender_mock"
+)
+
+const (
+	vpaInitializerConfigName = "vpa-initializer-config.k8s.io"
+	vpaInitializerName       = "vpa-initializer.k8s.io"
+)
+
+// Initializer initializes all pods matched by an existing vpa object
+// by assigning them recommended resource requests.
+type Initializer interface {
+	Run(stopCh <-chan struct{})
+}
+
+type initializer struct {
+	client          kubeclient.Interface
+	vpaLister       apimock.VerticalPodAutoscalerLister
+	podListerSynced cache.InformerSynced
+	informer        cache.SharedInformer
+	recommender     recommender.CachingRecommender
+	registerer      admissionregistrationv1alpha1.InitializerConfigurationInterface
+}
+
+func (i *initializer) Run(stopCh <-chan struct{}) {
```

Comment:
we should pass in context.Context to do cancellation.

Reply:
I haven't seen this in Kubernetes codebase, can you point me to an example?